UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/SalesCatalogueService.cs

```diff
@@ -0,0 +1,111 @@
+
+            logger.LogInformation(EventIds.SCSGetProductsFromSpecificDateRequestCompleted.ToEventId(),$"Get sales catalogue service from specific date time completed");
+            return response; 
+        }
+
+        public async Task<SalesCatalogueResponse> PostProductIdentifiersAsync(List<string> ProductIdentifiers)
+        {
+            logger.LogInformation(EventIds.SCSPostProductIdentifiersRequestStart.ToEventId(), $"Post sales catalogue service for ProductIdentifiers Started");
+
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(salesCatalogueConfig.Value.ResourceId);
+            var uri = $"/{salesCatalogueConfig.Value.Version}/productData/{salesCatalogueConfig.Value.ProductType}/products/productIdentifiers";
+
+            string payloadJson = JsonConvert.SerializeObject(ProductIdentifiers);
+
+            var httpResponse = await salesCatalogueClient.CallSalesCatalogueServiceApi(HttpMethod.Post, payloadJson, accessToken, uri);
+
+            SalesCatalogueResponse response = await CreateSalesCatalogueServiceResponse(httpResponse);
+
+            logger.LogInformation(EventIds.SCSPostProductIdentifiersRequestCompleted.ToEventId(), $"Post sales catalogue service for ProductIdentifiers completed");
+            return response;
+        }
+
+        public async Task<SalesCatalogueResponse> PostProductVersionsAsync(List<ProductVersionRequest> productVersions)
+        {
+            logger.LogInformation(EventIds.SCSPostProductVersionsRequestStart.ToEventId(), $"Post sales catalouge service for ProductVersions started");
+
+            var accessToken = await authTokenProvider.GetManagedIdentityAuthAsync(salesCatalogueConfig.Value.ResourceId);
+            var uri = $"/{salesCatalogueConfig.Value.Version}/productData/{salesCatalogueConfig.Value.ProductType}/products/productVersions";
+
+            string payloadJson = JsonConvert.SerializeObject(productVersions);
+
+            var httpResponse = await salesCatalogueClient.CallSalesCatalogueServiceApi(HttpMethod.Post, payloadJson, accessToken, uri);
+
+            SalesCatalogueResponse response = await CreateSalesCatalogueServiceResponse(httpResponse);
+            
+            logger.LogInformation(EventIds.SCSPostProductVersionsRequestCompleted.ToEventId(), $"Post sales catalogue service for ProductVersions completed");
+            return response;
+        }
+
+        private async Task<SalesCatalogueResponse> CreateSalesCatalogueServiceResponse(HttpResponseMessage httpResponse)
+        {
+            var response = new SalesCatalogueResponse();
+            var body = await httpResponse.Content.ReadAsStringAsync();
+            if (httpResponse.StatusCode != HttpStatusCode.OK && httpResponse.StatusCode != HttpStatusCode.NotModified)
+
+            {
+                logger.LogError(EventIds.SalesCatalogueNonOkResponse.ToEventId(), $"Sales catalougue service responded with {httpResponse.StatusCode} and message {body}");
```

Comment:
Please can we add details of SCS endpoint/url that was used in log.

Reply:
Added URI = method in log in 6850 PBI PR